packages/iov-tendermint-rpc/src/encodings.ts

```diff
@@ -156,8 +157,13 @@ export class Base64 {
 }
 
 export class DateTime {
-  public static decode(dateTimeString: DateTimeString): ReadonlyDate {
-    return Encoding.fromRfc3339(dateTimeString);
+  public static decode(dateTimeString: DateTimeString): ReadonlyDateWithNanoseconds {
+    const readonlyDate = Encoding.fromRfc3339(dateTimeString);
+    const nanosecondsMatch = dateTimeString.match(/\.(\d+)Z$/);
+    const nanoseconds = nanosecondsMatch ? nanosecondsMatch[1] : "";
+    // tslint:disable-next-line:no-object-mutation
+    (readonlyDate as any).nanoseconds = parseInt(nanoseconds.padEnd(9, "0"), 10);
```

Comment:
This duplicates data (fractional digits 1-3). In

```
    // fractional seconds match either undefined or a string like ".1", ".123456789"
    const milliSeconds = matches[7] ? Math.floor(+matches[7] * 1000) : 0;
```

position 4-9 are truncated. So I wonder if we want to only store position 4-9 in the extra field, to be interpreted as _nanoseconds after the value stored in readonlyDate_